jfsd/utils.py

```diff
@@ -375,16 +294,238 @@ def initialize_single_neighborlist(
+    
+    @jit
+    def add_thermal_noise(net_vel,brow):
+        net_vel = net_vel.at[0::3].add(brow[0::6])
+        net_vel = net_vel.at[1::3].add(brow[1::6])
+        net_vel = net_vel.at[2::3].add(brow[2::6])
+        return net_vel
+    
+    displacement, shift = space.periodic_general(
+        jnp.array([[L, 0., 0.], [0., L, 0.], [0., 0., L]]),fractional_coordinates=False)
+        
+    key = jrandom.PRNGKey(seed)
+        
+    T = 0.001
+    sigma = 2.15 #2.05 #particle diameter
+    net_vel = jnp.zeros(3*N)
+    brow_time = sigma*sigma / (4*T)
+    dt = brow_time / (1e4)
+    Nsteps = int(brow_time / dt)
+    key, random_coord = generate_random_array(key, (N*3))  
+    random_coord = (random_coord-1/2)*L
+    positions = jnp.zeros((N,3))
+    positions = positions.at[:,0].set(random_coord[0::3])
+    positions = positions.at[:,1].set(random_coord[1::3])
+    positions = positions.at[:,2].set(random_coord[2::3])
+        
+    displacements = (
+        space.map_product(displacement))(positions, positions)
+    
+    neighbor_fn = initialize_single_neighborlist(
+        2.2, L, L, L, displacement)
+    nbrs = allocate_nlist(positions+jnp.array([L, L, L])/2, neighbor_fn)
+    nl = np.array(nbrs.idx)
+    
+    overlaps = check_overlap(displacements,2.002)
+    buffer = (displacements[nl[0, :],nl[1, :],:])
+    buffer = jnp.sqrt(buffer[:,0]*buffer[:,0]+buffer[:,1]*buffer[:,1]+buffer[:,2]*buffer[:,2])
+    
+    k = 30*np.sqrt(6*T/(sigma*sigma*dt))  # spring constant
+    phi_eff = N/(L*L*L)*(sigma*sigma)*np.pi/3
+    print("Effective vol fraction is ", phi_eff)
+    if(phi_eff > 0.65):
+                raise ValueError(
+                    "Attempted to create particles configuration too dense. Use imported coordinates instead. Abort!")    
+    print("Overlaps while creating random configurations: ", overlaps, ' Steps to perform are ', Nsteps, ' k is ', k)    
+    while(overlaps>0):
+        start_time = time.time()
+
+        for i in range(Nsteps):
+            #Compute Velocity (Brownian + hard-sphere)
+            
+            #Compute distance vectors and neighbor list indices
+            (r, indices_i, indices_j) = precomputeBD(positions, nl, displacements, N, L, L, L)
+            
+            #compute forces for each pair
+            net_vel = compute_hardsphere(net_vel,displacements,indices_i,indices_j)
+
+            #Compute and add Brownian velocity
+            key, random_noise = generate_random_array(key, (6*N))
+            brow = thermal.compute_BD_randomforce(N, T, dt,random_noise)   
+            net_vel = add_thermal_noise(net_vel,brow)
+            
+            #Update positions
+            new_positions, nbrs, new_displacements, net_vel = update_pos(positions,displacements,net_vel,nbrs)
+            
+            if(nbrs.did_buffer_overflow):
+                print('its happening')
+                nbrs = allocate_nlist(positions, neighbor_fn)
+                nl = np.array(nbrs.idx)
```

Comment:
I guess this is a neighbor list, but it might be nicer to have a bit more descriptive variable names in this section.  Perhaps "k" is too little descriptive as well? Might be confused with k_b.

Reply:
Better documentation/description (and nomenclature) will be implemented soon. Also, in soft matter particle simulations 'k' is often used as a spring constant.